src/server/game/Entities/Player/Player.cpp

```diff
@@ -15760,15 +15760,20 @@ void Player::RewardQuest(Quest const* quest, LootItemType rewardType, uint32 rew
     uint32 XP = GetQuestXPReward(quest);
 
     int32 moneyRew = 0;
+    float moneyMod = 1.0f;
     if (getLevel() < sWorld->getIntConfig(CONFIG_MAX_PLAYER_LEVEL))
-        GiveXP(XP, nullptr);
+        GiveXP(XP * (HasWarmodeEnlistedAura() ? GetWarmodeAuraMod() : 1.0f), nullptr);
     else
+    {
         moneyRew = int32(quest->GetRewMoneyMaxLevel() * sWorld->getRate(RATE_DROP_MONEY));
+        moneyMod = GetWarmodeAuraMod();
```

Comment:
Should be implemented using SPELL_AURA_MOD_MONEY_GAIN inside `Player::GetQuestMoneyReward` (with `GetTotalAuraModifierByMiscValue(SPELL_AURA_MOD_MONEY_GAIN, 0)`)